### PR TITLE
DOC: Update README and add input descriptions

### PR DESCRIPTION
**Pull request by tbirdso**
Addresses https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/issues/28

### CODE REVIEW

**Comment by tbirdso** [comment]
The updated README with markdown can be viewed here: https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/blob/4f466d59cd207ee472d3563c3f19e9821935b667/README.md

**Comment by dzenanz** [comment]
I find it more useful to look at the [rich diff]([`4f466d5` (#30)](https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/pull/30/commits/4f466d59cd207ee472d3563c3f19e9821935b667)?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5).

**Comment by dzenanz** [comment]
Ah, I now see that almost the whole file is new 😄

**Comment by tbirdso** [comment]
> I find it more useful to look at the rich diff: [4f466d5?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5](https://github.com/InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/commit/4f466d59cd207ee472d3563c3f19e9821935b667?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

Great tip. I will do this in the future.

**Review by jhlegarreta — COMMENTED** [review]
Thanks for doing this Tom. Very helpful.

A few comments:
- Can the changes to the workflows and the `README` file be split into two different commits?
- Use the Markdown syntax to create links instead of displaying the raw URLs. May make the text more readable.
- Maybe a matter of preference, but I'd write GitHub to match the case they use to refer to it. Same goes for macOS.
- Add a link to PyPI.
- Maybe you can shorten GitHub Actions as GHA after the first mention.
- Add a note to the LICENSE.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,168 @@
...
+for the majority of ITK C++ external modules. However, it may not be feasible to adapt
+reusable workflows to fit every project's needs, particularly in the case where
+build setup requires fetching project-specific third party dependencies before building.
+
+If your project requires build steps not currently addressed by `ITKRemoteModuleBuildTestPackageAction`
+then consider doing one of the following:
+1. Implement build setup steps in your project's CMake procedures in `CMakeLists.txt` such that
+   building with CMake in a reusable workflow handles extraneous dependency retrieval steps.
+2. Fork the workflows in `ITKRemoteModuleBuildTestPackageAction` for your project and manually
+   replicating subsequent updates in your workflow;
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply README.md]: Fixed

**Comment by tbirdso** [comment]
@jhlegarreta Thanks for the feedback. I think I've addressed all of the points you noted, please let me know if anything is missing.

**Comment by jhlegarreta** [comment]
> @jhlegarreta Thanks for the feedback. I think I've addressed all of the points you noted, please let me know if anything is missing.

Looks good. Thanks Tom. Squash and merge the WIP commits into the README DOC commit and it'd be ready to go for me.

**Comment by tbirdso** [comment]
Squashed changes

**Comment by thewtex** [comment]
@tbirdso thanks!

**Pull request merged**